cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/ClusterModel.java

```diff
@@ -714,6 +716,57 @@ public Replica createReplica(String rackId, int brokerId, TopicPartition tp, int
     return replica;
   }
 
+  /**
+   * Add an imaginary replica to the cluster.The replica does not correspond to any existing replica in the cluster, but
+   * a replica we are going to add to the cluster. The imaginary replica's original broker will not be any existing broker
+   * so that it will be treated as an immigrant replica for whatever broker it is assigned to and grant goals greatest
+   * freedom to allocate to an existing broker.
+   *
+   * @param topicPartition The topic partition of the imaginary replica.
+   * @param brokerId       Id of broker which the imaginary replica is assigned to.
+   */
+  public void addReplica(TopicPartition topicPartition, int brokerId) {
```

Comment:
I think I already added the replica to `rack`/`host` [here](https://github.com/linkedin/cruise-control/blob/ef195916f9d90f3dde428253ce03641f497ff03c/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/ClusterModel.java#L290). Note `rack.addReplica()` will call `host.addReplica()` and `host.addReplica()` will call  `broker. addReplica()`, in this way we add the load to all the place.

The reason why I do not modify `createReplica` to support the usage here is because here we need the logic from both [createReplica](createReplica) and [setReplicaLoad](setReplicaLoad), maybe we can first merge the logic of these two method, then modify the merged to method to support our use case here.